test/JoeLensV2.js

```diff
@@ -0,0 +1,260 @@
+      "JoeLens"
+    );
+    this.JoeLensCFNew = await ethers.getContractFactory(
+      "JoeLensV2"
+    );
+    this.JoetrollerCF = await ethers.getContractFactory("Joetroller");
+    this.JoeCF = await ethers.getContractFactory("JErc20");
+    this.RewardDistributorCF = await ethers.getContractFactory(
+      "RewardDistributorV2"
+    );
+    this.RewardLensCF = await ethers.getContractFactory("RewardLens")
+    this.JUsdcDelegateCF = await ethers.getContractFactory(
+      JUSDC_DELEGATE_ARTIFACT.abi,
+      JUSDC_DELEGATE_ARTIFACT.bytecode
+    );
+    this.PriceOracleCF = await ethers.getContractFactory(
+      "PriceOracleProxyUSD"
+    );
+
+    // Contracts
+    this.jUsdcNative = await this.JUsdcDelegateCF.attach(
+      JUSDC_DELEGATOR_ARTIFACT.address
+    );
+    this.joetroller = await this.JoetrollerCF.attach(
+      UNITROLLER_ARTIFACT.address
+    );
+    this.joe = await this.JoeCF.attach(JOE_ADDRESS);
+  });
+
+  beforeEach(async function () {
+    // We reset the state before each tests
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [
+        {
+          forking: {
+            jsonRpcUrl: "https://api.avax.network/ext/bc/C/rpc",
+            blockNumber: 12947847,
+          },
+          live: false,
+          saveDeployments: false,
+          tags: ["test", "local"],
+        },
+      ],
+    });
+
+    // Impersonate Timelock address, which is the owner of lending contracts
+    await network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [MULTISIG_ADDRESS],
+    });
+    this.admin = await ethers.getSigner(MULTISIG_ADDRESS);
+
+    // Impersonate Dev address
+    await network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [DEV_ADDRESS],
+    });
+    this.dev = await ethers.getSigner(DEV_ADDRESS);
+
+    // Fund dev with AVAX
+    await this.alice.sendTransaction({
+      to: this.dev.address,
+      value: ethers.utils.parseEther("10"),
+    });
+
+    // Deploy rewarder and fund with JOE
+    this.rewardDistributor = await upgrades.deployProxy(
+      this.RewardDistributorCF,
+      []
+    );
+
+    await this.joe.connect(this.dev).transfer(
+      this.rewardDistributor.address,
+      ethers.utils.parseEther("1000000") // 1 million JOE
+    );
+
+    await this.rewardDistributor.setAdmin(this.dev.address);
+
+    await this.rewardDistributor.connect(this.dev).setJoe(this.joe.address);
+    await this.joetroller
+      .connect(this.admin)
+      ._setRewardDistributor(this.rewardDistributor.address);
+
+    // Deploy Reward Lens for old joe lens and set rewards 
+    this.rewardLens = await this.RewardLensCF.deploy();
+    await this.rewardLens.deployed(); 
+
+    await this.rewardLens.setMarketRewards(
+      this.jUsdcNative.address, 
+      ethers.utils.parseEther("0.1"), 
+      ethers.utils.parseEther("0.3"), 
+      ethers.utils.parseEther("0.1"), 
+      ethers.utils.parseEther("0.3")
+    );
```

Comment:
Confused as to why we need to set reward speed on RewardLens. Feel like we can remove.

Reply:
I did this so we could check that the reward speeds from JoeLensV1 match the reward speeds from JoeLensV2 if they're both set to something in RewardLens/RewardDistributorV2. But can remove this since you said its not needed